src/axios-har-tracker.ts

```diff
@@ -1,112 +1,204 @@
   request: {},
   response: {},
-  startedDateTime: startDate,
-  time: -1,
+  startedDateTime: string,
+  time: number,
   cache: {},
   timings: {
-    blocked: -1,
-    dns: -1,
-    ssl: -1,
-    connect: -1,
-    send: 10,
-    wait: 10,
-    receive: 10,
-    _blocked_queueing: -1
+    blocked: number,
+    dns: number,
+    ssl: number,
+    connect: number,
+    send: number,
+    wait: number,
+    receive: number,
+    _blocked_queueing: number
   }
 };
 
-async function axiosTracker(call) {
-  axios.interceptors.request.use(
-    async config => {
-      config.validateStatus = function () {
-        return true;
-      };
-      config.headers['request-startTime'] = process.hrtime();
-      const fullCookie = JSON.stringify(config.headers['Cookie']);
-      const version = config.httpVersion === undefined ? 'HTTP/1.1' : 'HTTP/' + config.httpVersion;
-
-      newEntry.request = {
-        method: config.method,
-        url: config.url,
-        httpVersion: version,
-        cookies: getCookies(fullCookie),
+export class AxiosHarTracker {
+
+  private axios: AxiosStatic;
+  private generatedHar: HarFile;
+  private newEntry: NewEntry;
+  private date = new Date().toISOString();
+
+  private requestObject(config) {
+
+    config.headers['request-startTime'] = process.hrtime();
+    let requestObject = {
+      method: config.method,
+      url: config.url,
+      httpVersion: 'HTTP/1.1',
+      cookies: this.getCookies(JSON.stringify(config.headers['Cookie'])),
+      headers: this.getHeaders(config.headers['common']),
+      queryString: this.getParams(config.params),
+      headersSize: -1,
+      bodySize: -1
+    };
+    return requestObject;
+  }
+
+  private responseObject(response) {
+    if (response) {
+      let responseObject = {
+        status: response.status,
+        statusText: response.statusText,
+        headers: this.getHeaders(response.headers),
+        startedDateTime: new Date(response.headers.date),
+        time: response.headers['request-duration'] = Math.round(
+          process.hrtime(response.headers['request-startTime'])[0] * 1000 +
+          process.hrtime(response.headers['request-startTime'])[1] / 1000000
+        ),
+        httpVersion: 'HTTP/1.1',
+        cookies: this.getCookies(JSON.stringify(response.config.headers['Cookie'])),
+        bodySize: JSON.stringify(response.data).length,
+        redirectURL: '',
+        headersSize: -1,
+        content: {
+          size: JSON.stringify(response.data).length,
+          mimeType: response.headers['content-type'] ? response.headers['content-type'] : 'text/plain',
+          text: JSON.stringify(response.data)
+        },
+        cache: {},
+        timings: {
+          blocked: -1,
+          dns: -1,
+          ssl: -1,
+          connect: -1,
+          send: 10,
+          wait: 10,
+          receive: 10,
+          _blocked_queueing: -1
+        }
+      }
+      return responseObject;
+    } else {
+      let responseObject = {
```

Comment:
Instead of defining the same object structure in `if` and `else`, I would prefer to handle it in one object with conditions for those fields that need `response` object